frontend/src/metabase/dashboard/selectors.ts

```diff
@@ -321,12 +354,41 @@ export const getMissingRequiredParameters = createSelector(
     ),
 );
 
-export const getParameterMappingOptions = createSelector(
-  [getMetadata, getEditingParameter, getCard, getDashCard],
-  (metadata, parameter, card, dashcard) => {
-    return _getParameterMappingOptions(metadata, parameter, card, dashcard);
+export const getQuestionByDashcardMemo = createCachedSelector(
+  [
+    (_state: State, props: { dashcard: DashboardCard }) => props.dashcard,
+    getMetadata,
+  ],
+  (dashcard, metadata) => {
+    return new Question(dashcard.card, metadata);
   },
-);
+)((_state, props) => {
+  return props.dashcard.id;
+});
+
+export const getParameterMappingOptions = createCachedSelector(
+  [
+    getMetadata,
+    getEditingParameter,
+    getCard,
+    getDashCard,
+    getQuestionByDashcardMemo,
+  ],
+  (metadata, parameter, card, dashcard, q) => {
+    let question;
+
+    if (dashcard.card_id) {
+      question = q;
+    } else {
+      // TODO: probably we can remove it as dashcards are text or heading or action?
+      question = new Question(card, metadata);
+    }
+
+    return _getParameterMappingOptions(question, parameter, card, dashcard);
```

Comment:
Yeah, we might want to use `isVirtualDashCard` here to skip heading, text, link cards, etc.
I can't recall if we're using this code for actions, though. It's definitely possible to map dashboard filter values to action parameters, just not sure which code is doing that

Reply:
as far as I understand, we don't need to create a question and pass question for virtual card and action card